pragma-node/src/handlers/entries/subscribe_to_entry.rs

```diff
@@ -0,0 +1,260 @@
+};
+use crate::utils::{get_encoded_pair_id, get_entry_hash, HashError};
+use crate::AppState;
+
+use super::{AssetOraclePrice, SignedPublisherPrice};
+
+/// "PRAGMA" to number is bigger than 2**40 - we alias it to "PRGM" to fit in 40 bits.
+/// Needed for StarkEx signature.
+/// See:
+/// https://docs.starkware.co/starkex/perpetual/becoming-an-oracle-provider-for-starkex.html
+const PRAGMA_ORACLE_NAME: &str = "PRGM";
+
+/// Interval in milliseconds that the channel will update the client with the latest prices.
+const CHANNEL_UPDATE_INTERVAL_IN_MS: u64 = 500;
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+enum SubscriptionType {
+    #[serde(rename = "subscribe")]
+    #[default]
+    Subscribe,
+    #[serde(rename = "unsubscribe")]
+    Unsubscribe,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SubscriptionRequest {
+    msg_type: SubscriptionType,
+    pairs: Vec<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SubscriptionAck {
+    msg_type: SubscriptionType,
+    pairs: Vec<String>,
+}
+
+#[utoipa::path(
+    get,
+    path = "/node/v1/data/subscribe",
+    responses(
+        (
+            status = 200,
+            description = "Subscribe to a list of entries",
+            body = [SubscribeToEntryResponse]
+        )
+    )
+)]
+pub async fn subscribe_to_entry(
+    ws: WebSocketUpgrade,
+    State(state): State<AppState>,
+) -> impl IntoResponse {
+    ws.on_upgrade(move |socket| handle_channel(socket, state))
+}
+
+/// TODO(akhercha): currently only returns index median - need to add mark price
+/// Handle the WebSocket channel.
+async fn handle_channel(mut socket: WebSocket, state: AppState) {
+    let waiting_duration = Duration::from_millis(CHANNEL_UPDATE_INTERVAL_IN_MS);
+    let mut update_interval = interval(waiting_duration);
+    let mut subscribed_pairs: Vec<String> = Vec::new();
+
+    loop {
+        tokio::select! {
+            Some(msg) = socket.recv() => {
+                if let Ok(Message::Text(text)) = msg {
+                    handle_message_received(&mut socket, &state, &mut subscribed_pairs, text).await;
+                }
+            },
+            _ = update_interval.tick() => {
+                match send_median_entries(&mut socket, &state, &subscribed_pairs).await {
+                    Ok(_) => {},
+                    Err(_) => break
+                };
+            }
+        }
+    }
+}
+
+/// Handle the message received from the client.
+/// Subscribe or unsubscribe to the pairs requested.
+async fn handle_message_received(
+    socket: &mut WebSocket,
+    state: &AppState,
+    subscribed_pairs: &mut Vec<String>,
+    message: String,
+) {
+    if let Ok(subscription_msg) = serde_json::from_str::<SubscriptionRequest>(&message) {
+        match subscription_msg.msg_type {
+            SubscriptionType::Subscribe => {
+                let existing_pairs =
+                    only_existing_pairs(&state.timescale_pool, subscription_msg.pairs).await;
```

Comment:
what happens when the consumer tries to subscribe to an unexisting pair ?

Reply:
It's simply ignored - the server returns a `SubscriptionAck` messages that returns the currently subscribed pairs so the client can see which pairs successfully got registered

Reply:
ok lets not forget to add this in the docs